options/configurable.cc

```diff
@@ -239,9 +241,15 @@ Status Configurable::ParseOption(const ConfigOptions& config_options,
                                  const OptionTypeInfo& opt_info,
                                  const std::string& opt_name,
                                  const std::string& opt_value, void* opt_ptr) {
-  if (opt_info.IsMutable() || opt_info.IsConfigurable()) {
-    return opt_info.Parse(config_options, opt_name, opt_value, opt_ptr);
-  } else if (prepared_) {
+  if (opt_info.IsMutable()) {
+    if (config_options.only_mutable_options) {
+      ConfigOptions copy = config_options;
+      copy.only_mutable_options = false;
```

Comment:
Can you explain this section of code. If ```config_options.only_mutable_options``` is set true why does```copy.only_mutable_options = false```.  Or may be add comments in the code.

Reply:
I will do both (push a comment in the next update).  

There are two things of interest used repeatedly.  One is whether or not the field is marked as IsMutable.  Another is if we are operating on only mutable options or not.

If a field is marked as Mutable, then all of the fields in it should inherit its mutable property even if they are not explicitly marked as mutable.   In other words, if a field is marked as mutable, we should toggle whether or not we are examining only mutable options. 

For example, if an object "o" has a member in it "Configurable c" and "c" is marked as Mutable, then all of the elements of "c" (when accessed from "o") will also be mutable even if they are not marked as such explicitly.

Reply:
ahh.. "One is whether or not the field is marked as IsMutable. Another is if we are operating on only mutable options or not." This makes sense. Thanks for explaining it.